### ISSUE TITLE
Is it possible to run PotreeConverter using Linux?

### ISSUE THREAD

**Issue by umbe1987** [potree/PotreeConverter#180]
Hi. I am following the instructions in the README.md on how to build PotreeConverter using Linux and gcc4.9. After I followed the instructions, I am not able to run PotreeConverter. Where am I supposed to run it from? In the terminal, if I simply type PotreeConverter, nothing happens.
Moreover, at the very final command in the LAStools build instructions ("make"), I received the following warnings (maybe something went wrong? Because the build instructions of PotreeConverter went fine...):

[100%] Building CXX object src/CMakeFiles/laszip.dir/laszip_dll.cpp.o
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp: In function ‘laszip_I32 laszip_open_writer(laszip_POINTER, const laszip_CHAR_, laszip_BOOL)’:
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp:1609:219: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘laszip_U64 {aka long long unsigned int}’ [-Wformat=]
           sprintf(laszip_dll->error, "inconsistent number_of_point_records %u and extended_number_of_point_records %I64d", laszip_dll->header.number_of_point_records, laszip_dll->header.extended_number_of_point_records);
                                                                                                                                                                                                                           ^
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp:1623:253: warning: format ‘%d’ expects argument of type ‘int’, but argument 6 has type ‘laszip_U64 {aka long long unsigned int}’ [-Wformat=]
             sprintf(laszip_dll->error, "inconsistent number_of_points_by_return[%u] %u and extended_number_of_points_by_return[%u] %I64d", i, laszip_dll->header.number_of_points_by_return[i], i, laszip_dll->header.extended_number_of_points_by_return[i]);
                                                                                                                                                                                                                                                             ^
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp:1646:188: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘laszip_U64 {aka long long unsigned int}’ [-Wformat=]
           sprintf(laszip_dll->error, "extended_number_of_point_records of %I64d is too much for 32-bit counters of compatibility mode", laszip_dll->header.extended_number_of_point_records);
                                                                                                                                                                                            ^
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp:1812:134: warning: format ‘%f’ expects argument of type ‘double’, but argument 3 has type ‘I32 {aka int}’ [-Wformat=]
                 sprintf(laszip_dll->error, "cannot add unknown U8 attribute %f of %d to attributer", i, number_of_existing_extrabytes);
                                                                                                                                      ^
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp: In function ‘laszip_I32 laszip_open_reader(laszip_POINTER, const laszip_CHAR_, laszip_BOOL*)’:
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp:3600:207: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 4 has type ‘U32 {aka unsigned int}’ [-Wformat=]
               fprintf(stderr,"WARNING: number_of_point_records is %u. but extended_number_of_point_records is %lld.\n", laszip_dll->header.number_of_point_records, number_of_extended_variable_length_records);
                                                                                                                                                                                                               ^
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp:3613:228: warning: format ‘%d’ expects argument of type ‘int’, but argument 5 has type ‘U64 {aka long long unsigned int}’ [-Wformat=]
                 fprintf(stderr,"WARNING: number_of_points_by_return[%d] is %u. but extended_number_of_points_by_return[%d] is %lld.\n", laszip_dll->header.number_of_points_by_return[i], i, extended_number_of_points_by_return, i);
                                                                                                                                                                                                                                    ^
/home/umberto/lastools/master/LASzip/src/laszip_dll.cpp:3613:228: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 6 has type ‘U32 {aka unsigned int}’ [-Wformat=]
Linking CXX shared library liblaszip.so
[100%] Built target laszip

**Comment by eokeeffe** [comment]
It should end up in your build directory in PotreeConverter.
If you want to call it from the command line, edit your PATH to include the directory to the executable

**Comment by spatialhast** [comment]
Example command for installation on Ubuntu 14.04:

> Install Boost / cmake / build-essential
> $ sudo apt-get install libboost-dev
> $ sudo apt-get install libboost-system-dev libboost-thread-dev libboost-filesystem-dev libboost-program-options-dev libboost-regex-dev
> $ sudo apt-get install cmake
> $ sudo apt-get install build-essential
> 
> Install gcc/g++ 4.9
> $ sudo add-apt-repository ppa:ubuntu-toolchain-r/test
> $ sudo apt-get update
> $ sudo apt-get install gcc-4.9 g++-4.9 cpp-4.9
> $ gcc --version
> $ cd /usr/bin
> $ sudo rm gcc g++ cpp
> $ sudo ln -s gcc-4.9 gcc
> $ sudo ln -s g++-4.9 g++
> $ sudo ln -s cpp-4.9 cpp
> 
> Install LAStools
> $ git clone https://github.com/m-schuetz/LAStools.git
> $ cd LAStools/LASzip
> $ mkdir build && cd build
> $ cmake -DCMAKE_BUILD_TYPE=Release ..
> $ make
> 
> $ git clone https://github.com/potree/PotreeConverter.git
> $ cd PotreeConverter
> $ mkdir build && cd build
> $ sudo cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=~/LAStools/LASzip/dll -DLASZIP_LIBRARY=~/LAStools/LASzip/build/src/liblaszip.so ..
> $ sudo make && sudo make install
> 
> $ PotreeConverter -h

**Comment by dnuske** [comment]
@spatialhast everything worked but the last compile `sudo make && sudo make install` for PotreeConverter. It raises:
```
ubuntu@ip-172-31-15-98:~/code/PotreeConverter/build$ sudo make && sudo make install
Scanning dependencies of target PotreeConverter
[  8%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/SparseGrid.cpp.o
[ 16%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeConverter.cpp.o
/home/ubuntu/code/PotreeConverter/PotreeConverter/src/PotreeConverter.cpp: In function ‘void Potree::writeSources(std::__cxx11::string, std::vector<std::__cxx11::basic_string<char> >, std::vector<int>, std::vector<Potree::AABB>, std::__cxx11::string)’:
/home/ubuntu/code/PotreeConverter/PotreeConverter/src/PotreeConverter.cpp:255:19: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for(int i = 0; i < sourceFilenames.size(); i++){
                   ^
[ 25%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/LASPointWriter.cpp.o
[ 33%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/main.cpp.o
/home/ubuntu/code/PotreeConverter/PotreeConverter/src/main.cpp: In function ‘void printArguments(Arguments&)’:
/home/ubuntu/code/PotreeConverter/PotreeConverter/src/main.cpp:268:19: warning: unused variable ‘s’ [-Wunused-variable]
   for(const auto &s : a.source) {
                   ^
[ 41%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PTXPointReader.cpp.o
[ 50%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/stuff.cpp.o
/home/ubuntu/code/PotreeConverter/PotreeConverter/src/stuff.cpp:38:21: warning: extra tokens at end of #include directive
 #include <Windows.h>>
                     ^
/home/ubuntu/code/PotreeConverter/PotreeConverter/src/stuff.cpp:38:22: fatal error: Windows.h: No such file or directory
compilation terminated.
PotreeConverter/CMakeFiles/PotreeConverter.dir/build.make:182: recipe for target 'PotreeConverter/CMakeFiles/PotreeConverter.dir/src/stuff.cpp.o' failed
make[2]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/src/stuff.cpp.o] Error 1
CMakeFiles/Makefile2:85: recipe for target 'PotreeConverter/CMakeFiles/PotreeConverter.dir/all' failed
make[1]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by dnuske** [comment]
managed to compile it based on @spatialhast and help from @dcaminos .

what I did was, on file /PotreeConverter/src/stuff.cpp
 - removed the if block of BOOST_OS_WINDOWS from all places,
 - included `#include <linux/limits.h>`  on top of it

then it compiled with `sudo cmake` then `sudo make` then `sudo make install` 

After that just had to include the file `liblaszip.so` location in the path and then
`$ PotreeConverter -h` just worked.

**Comment by themmes** [comment]
@dnuske After your two suggestions still getting errors:

```
sudo cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=/home/tom/synth/LAStools/LASzip/build/src/liblaszip.so -DLASZIP_INCLUDE_DIRS=/home/tom/synth/LAStools/LASzip/dll
...
tom@vision:~/synth/PotreeConverter-1.5RC$ sudo make && sudo make install
Scanning dependencies of target PotreeConverter
[  8%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/SparseGrid.cpp.o
[ 16%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeWriter.cpp.o
/home/tom/synth/PotreeConverter-1.5RC/PotreeConverter/src/PotreeWriter.cpp: In lambda function:
/home/tom/synth/PotreeConverter-1.5RC/PotreeConverter/src/PotreeWriter.cpp:316:35: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if(append && (writer->numPoints != this->numAccepted)){
                                   ^
[ 25%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PointAttributes.cpp.o
[ 33%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PTXPointReader.cpp.o
[ 41%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/BINPointReader.cpp.o
[ 50%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/LASPointReader.cpp.o
[ 58%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/stuff.cpp.o
/home/tom/synth/PotreeConverter-1.5RC/PotreeConverter/src/stuff.cpp: In function ‘std::__cxx11::string Potree::getExecutablePath()’:
/home/tom/synth/PotreeConverter-1.5RC/PotreeConverter/src/stuff.cpp:207:15: error: ‘MAX_PATH’ was not declared in this scope
  char  buffer[MAX_PATH];
               ^
/home/tom/synth/PotreeConverter-1.5RC/PotreeConverter/src/stuff.cpp:208:27: error: ‘buffer’ was not declared in this scope
  GetModuleFileName( NULL, buffer, MAX_PATH );
                           ^
/home/tom/synth/PotreeConverter-1.5RC/PotreeConverter/src/stuff.cpp:208:44: error: ‘GetModuleFileName’ was not declared in this scope
  GetModuleFileName( NULL, buffer, MAX_PATH );
                                            ^
PotreeConverter/CMakeFiles/PotreeConverter.dir/build.make:206: recipe for target 'PotreeConverter/CMakeFiles/PotreeConverter.dir/src/stuff.cpp.o' failed
make[2]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/src/stuff.cpp.o] Error 1
CMakeFiles/Makefile2:85: recipe for target 'PotreeConverter/CMakeFiles/PotreeConverter.dir/all' failed
make[1]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

Any clue on how to fix these? Running Ub 16.04 Xenial

**Comment by ahmadshabbir2468** [comment]
How can i Install Potree Converter on fresh Ubuntu ? is there any step by step solution ?Because i try a lot and nothing works Fine

**Comment by denizhosgor** [comment]
On Ubuntu 16.04.3

 apt-get install python-software-properties git
 apt-get install build-essential cmake g++
 apt-get install libboost-all-dev
 apt-get install cmake-curses-gui
 apt-get install gcc

git clone https://github.com/m-schuetz/LAStools.git master
...
...

git clone https://github.com/potree/PotreeConverter.git master
...
...

**Comment by spatialhast** [comment]
@denizhosgor thanks for the instructions, today I tested your installation commands on Ubuntu Server 16.04.3 - all works fine.

```
$ sudo apt-get install python-software-properties git
$ sudo apt-get install build-essential cmake g++
$ sudo apt-get install libboost-all-dev
$ sudo apt-get install cmake-curses-gui
$ sudo apt-get install gcc
```

LAStools:
```
$ git clone https://github.com/m-schuetz/LAStools.git
$ cd LAStools/LASzip
$ mkdir build && cd build
$ cmake -DCMAKE_BUILD_TYPE=Release ..
$ make
```

Potree Converter:
```
$ git clone https://github.com/potree/PotreeConverter.git
$ cd PotreeConverter
$ mkdir build && cd build
$ sudo cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=/home/opengeo/LAStools/LASzip/dll -DLASZIP_LIBRARY=/home/opengeo/LAStools/LASzip/build/src/liblaszip.so ..
$ sudo make && sudo make install
$ sudo ln -s ~/LAStools/LASzip/build/src/liblaszip.so /usr/lib
$ PotreeConverter -h
```

**Comment by ahmadshabbir2468** [comment]
Thanks  Its Works PERFECTLY Fine

**Comment by boopathit98** [comment]
i will use all step follow install build generated and PotreeConverter -h cmd i will type not found error show how to solve 

i used OS as ubuntu16.04

**Comment by rhurlbatt** [comment]
@denizhosgor can you explain how to correct the following error while doing sudo make and install for PotreeConverter

rick@rick-x360:~/PotreeConverter/build$ sudo make && sudo make install
Scanning dependencies of target PotreeConverter
[  8%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/SparseGrid.cpp.o
[ 16%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeWriter.cpp.o
In file included from /home/rick/PotreeConverter/PotreeConverter/src/PotreeWriter.cpp:19:0:
/home/rick/PotreeConverter/PotreeConverter/include/LASPointReader.h:10:24: fatal error: laszip_api.h: No such file or directory
compilation terminated.
PotreeConverter/CMakeFiles/PotreeConverter.dir/build.make:86: recipe for target 'PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeWriter.cpp.o' failed
make[2]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeWriter.cpp.o] Error 1
CMakeFiles/Makefile2:85: recipe for target 'PotreeConverter/CMakeFiles/PotreeConverter.dir/all' failed
make[1]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by Gwen22200** [comment]
Hello,

I have the same probleme to install potree converter on ubuntu 14.04 with this command : 

git clone https://github.com/potree/PotreeConverter.git && cd PotreeConverter && mkdir build && cd build && sudo cmake -DCMAKE_BUILD_TYPE=Release  DLASZIP_INCLUDE_DIRS=/home/tools/LAStools/LASzip/dll -DLASZIP_LIBRARY=/home/tools/LAStools/LASzip/build/src/liblaszip.so .. && sudo make && sudo make install

-----------------------
-- result--------------
-----------------------
[ 54%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeWriter.cpp.o
In file included from /home/tools/LAStools/LASzip/build/PotreeConverter/build/PotreeConverter/PotreeConverter/src/PotreeWriter.cpp:19:0:
/home/tools/LAStools/LASzip/build/PotreeConverter/build/PotreeConverter/PotreeConverter/include/LASPointReader.h:10:24: fatal error: laszip_api.h: Aucun fichier ou dossier de ce type
compilation terminated.
make[2]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeWriter.cpp.o] Erreur 1
make[1]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/all] Erreur 2
make: *** [all] Erreur 2

When i go into cd /home/tools/LAStools/LASzip/build/PotreeConverter/build/PotreeConverter/PotreeConverter/include/ no file name laszip_api.h, i try to add it manually but no change.

Thank you for your help

Gwen

**Comment by flightsurvey** [comment]
I had the same problem but manually copied the file in. This happened when I ran the last few commands:

paddy@paddy-Vostro-320:~/PotreeConverter/build$ sudo cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=/home/opengeo/LAStools/LASzip/dll -DLASZIP_LIBRARY=/home/opengeo/LAStools/LASzip/build/src/liblaszip.so ..
-- Configuring done
-- Generating done
-- Build files have been written to: /home/paddy/PotreeConverter/build
paddy@paddy-Vostro-320:~/PotreeConverter/build$ sudo cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=~/dev/workspaces/lastools/master/LASzip/dll -DLASZIP_LIBRARY=~/dev/workspaces/lastools/master/LASzip/build/src/liblaszip.so ..
-- Configuring done
-- Generating done
-- Build files have been written to: /home/paddy/PotreeConverter/build
paddy@paddy-Vostro-320:~/PotreeConverter/build$ sudo make
Scanning dependencies of target PotreeConverter
[  8%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PTXPointReader.cpp.o
[ 16%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/LASPointWriter.cpp.o
[ 25%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/stuff.cpp.o
/home/paddy/PotreeConverter/PotreeConverter/src/stuff.cpp: In function ‘std::vector<std::__cxx11::basic_string<char> > Potree::split(std::__cxx11::string, std::vector<char>)’:
/home/paddy/PotreeConverter/PotreeConverter/src/stuff.cpp:252:20: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (int i = 0; i < str.size(); i++) {
                    ^
/home/paddy/PotreeConverter/PotreeConverter/src/stuff.cpp:263:12: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  if (start < str.size()) {
            ^
[ 33%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/GridCell.cpp.o
[ 41%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/SparseGrid.cpp.o
[ 50%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeWriter.cpp.o
/home/paddy/PotreeConverter/PotreeConverter/src/PotreeWriter.cpp: In lambda function:
/home/paddy/PotreeConverter/PotreeConverter/src/PotreeWriter.cpp:315:35: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if(append && (writer->numPoints != this->numAccepted)){
                                   ^
[ 58%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/LASPointReader.cpp.o
[ 66%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PointAttributes.cpp.o
[ 75%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/BINPointReader.cpp.o
[ 83%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/PotreeConverter.cpp.o
/home/paddy/PotreeConverter/PotreeConverter/src/PotreeConverter.cpp: In function ‘void Potree::writeSources(std::__cxx11::string, std::vector<std::__cxx11::basic_string<char> >, std::vector<int>, std::vector<Potree::AABB>, std::__cxx11::string)’:
/home/paddy/PotreeConverter/PotreeConverter/src/PotreeConverter.cpp:268:19: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for(int i = 0; i < sourceFilenames.size(); i++){
                   ^
[ 91%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/main.cpp.o
In file included from /home/paddy/PotreeConverter/PotreeConverter/src/main.cpp:13:0:
/home/paddy/PotreeConverter/PotreeConverter/lib/arguments/arguments.hpp: In member function ‘std::vector<std::__cxx11::basic_string<char> > Argument::split(std::__cxx11::string, std::vector<char>)’:
/home/paddy/PotreeConverter/PotreeConverter/lib/arguments/arguments.hpp:89:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < str.size(); i++) {
                     ^
/home/paddy/PotreeConverter/PotreeConverter/lib/arguments/arguments.hpp:102:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (start < str.size()) {
             ^
/home/paddy/PotreeConverter/PotreeConverter/lib/arguments/arguments.hpp: In member function ‘std::__cxx11::string Arguments::usage()’:
/home/paddy/PotreeConverter/PotreeConverter/lib/arguments/arguments.hpp:259:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < argdefs.size(); i++) {
                     ^
/home/paddy/PotreeConverter/PotreeConverter/src/main.cpp: In function ‘void printArguments(PotreeArguments&)’:
/home/paddy/PotreeConverter/PotreeConverter/src/main.cpp:257:19: warning: unused variable ‘s’ [-Wunused-variable]
   for(const auto &s : a.source) {
                   ^
make[2]: *** No rule to make target '/home/paddy/dev/workspaces/lastools/master/LASzip/build/src/liblaszip.so', needed by 'PotreeConverter/PotreeConverter'. Stop.
CMakeFiles/Makefile2:85: recipe for target 'PotreeConverter/CMakeFiles/PotreeConverter.dir/all' failed
make[1]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
paddy@paddy-Vostro-320:~/PotreeConverter/build$ ^C
paddy@paddy-Vostro-320:~/PotreeConverter/build$

**Comment by Nastasija94** [comment]
When I run cmake -DCMAKE_BUILD_TYPE=Release .. I get:

-- The CXX compiler identification is Clang 6.0.0
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- broken
CMake Error at /usr/share/cmake-3.10/Modules/CMakeTestCXXCompiler.cmake:45 (message):
  The C++ compiler

    "/usr/bin/clang++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/user/LAStools/LASzip/build/CMakeFiles/CMakeTmp
    
    Run Build Command:"/usr/bin/make" "cmTC_1b9ee/fast"
    /usr/bin/make -f CMakeFiles/cmTC_1b9ee.dir/build.make CMakeFiles/cmTC_1b9ee.dir/build
    make[1]: Entering directory '/home/user/LAStools/LASzip/build/CMakeFiles/CMakeTmp'
    Building CXX object CMakeFiles/cmTC_1b9ee.dir/testCXXCompiler.cxx.o
    /usr/bin/clang++     -o CMakeFiles/cmTC_1b9ee.dir/testCXXCompiler.cxx.o -c /home/user/LAStools/LASzip/build/CMakeFiles/CMakeTmp/testCXXCompiler.cxx
    Linking CXX executable cmTC_1b9ee
    /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_1b9ee.dir/link.txt --verbose=1
    /usr/bin/clang++       -rdynamic CMakeFiles/cmTC_1b9ee.dir/testCXXCompiler.cxx.o  -o cmTC_1b9ee 
    /usr/bin/ld: cannot find -lstdc++
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    CMakeFiles/cmTC_1b9ee.dir/build.make:97: recipe for target 'cmTC_1b9ee' failed
    make[1]: *** [cmTC_1b9ee] Error 1
    make[1]: Leaving directory '/home/user/LAStools/LASzip/build/CMakeFiles/CMakeTmp'
    Makefile:126: recipe for target 'cmTC_1b9ee/fast' failed
    make: *** [cmTC_1b9ee/fast] Error 2
    
  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:3 (project)

-- Configuring incomplete, errors occurred!
See also "/home/user/LAStools/LASzip/build/CMakeFiles/CMakeOutput.log".
See also "/home/user/LAStools/LASzip/build/CMakeFiles/CMakeError.log".

Does someone have a clue why I get this? :/

**Comment by raymondwm** [comment]
make
[ 58%] Built target brotlicommon-static
[ 58%] Built target brotlienc-static
[ 58%] Built target laszip
[ 58%] Built target brotlidec-static
[ 58%] Building CXX object CMakeFiles/PotreeConverter.dir/Converter/src/chunker_countsort_laszip.cpp.o
In file included from /Users/rawang/pythonProject/PotreeConverter/./Converter/include/Attributes.h:10,
                 from /Users/rawang/pythonProject/PotreeConverter/./Converter/include/chunker_countsort_laszip.h:8,
                 from /Users/rawang/pythonProject/PotreeConverter/Converter/src/chunker_countsort_laszip.cpp:10:
/Users/rawang/pythonProject/PotreeConverter/./Converter/modules/unsuck/unsuck.hpp: In function 'std::vector<unsigned char> readBinaryFile(std::string, uint64_t, uint64_t)':
/Users/rawang/pythonProject/PotreeConverter/./Converter/modules/unsuck/unsuck.hpp:435:17: error: 'fseek_64_all_platforms' was not declared in this scope
  435 |                 fseek_64_all_platforms(file, start, SEEK_SET);
      |                 ^~~~~~~~~~~~~~~~~~~~~~
/Users/rawang/pythonProject/PotreeConverter/./Converter/modules/unsuck/unsuck.hpp:444:17: error: 'fseek_64_all_platforms' was not declared in this scope
  444 |                 fseek_64_all_platforms(file, start, SEEK_SET);
      |                 ^~~~~~~~~~~~~~~~~~~~~~
/Users/rawang/pythonProject/PotreeConverter/./Converter/modules/unsuck/unsuck.hpp: In function 'void readBinaryFile(std::string, uint64_t, uint64_t, void*)':
/Users/rawang/pythonProject/PotreeConverter/./Converter/modules/unsuck/unsuck.hpp:462:17: error: 'fseek_64_all_platforms' was not declared in this scope
  462 |                 fseek_64_all_platforms(file, start, SEEK_SET);
      |                 ^~~~~~~~~~~~~~~~~~~~~~
/Users/rawang/pythonProject/PotreeConverter/./Converter/modules/unsuck/unsuck.hpp:466:17: error: 'fseek_64_all_platforms' was not declared in this scope
  466 |                 fseek_64_all_platforms(file, start, SEEK_SET);
      |                 ^~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/rawang/pythonProject/PotreeConverter/Converter/src/chunker_countsort_laszip.cpp:19:
/Users/rawang/pythonProject/PotreeConverter/./Converter/libs/json/json.hpp: In static member function 'static CharType nlohmann::detail::binary_writer<BasicJsonType, CharType>::to_char_type(uint8_t)':
/Users/rawang/pythonProject/PotreeConverter/./Converter/libs/json/json.hpp:12595:28: warning: 'template<class _Tp> struct std::is_pod' is deprecated: use is_standard_layout && is_trivial instead [-Wdeprecated-declarations]
12595 |         static_assert(std::is_pod<CharType>::value, "CharType must be POD");
      |                            ^~~~~~
In file included from /opt/homebrew/Cellar/gcc@11/11.4.0/include/c++/11/bits/move.h:57,
                 from /opt/homebrew/Cellar/gcc@11/11.4.0/include/c++/11/bits/exception_ptr.h:43,
                 from /opt/homebrew/Cellar/gcc@11/11.4.0/include/c++/11/exception:153,
                 from /opt/homebrew/Cellar/gcc@11/11.4.0/include/c++/11/ios:39,
                 from /opt/homebrew/Cellar/gcc@11/11.4.0/include/c++/11/ostream:38,
                 from /opt/homebrew/Cellar/gcc@11/11.4.0/include/c++/11/iostream:39,
                 from /Users/rawang/pythonProject/PotreeConverter/Converter/src/chunker_countsort_laszip.cpp:2:
/opt/homebrew/Cellar/gcc@11/11.4.0/include/c++/11/type_traits:733:5: note: declared here
  733 |     is_pod
      |     ^~~~~~
make[2]: *** [CMakeFiles/PotreeConverter.dir/Converter/src/chunker_countsort_laszip.cpp.o] Error 1
make[1]: *** [CMakeFiles/PotreeConverter.dir/all] Error 2
make: *** [all] Error 2
--------------------------------
再mac 上安装遇到以上问题
